### PR TITLE
✨Create fun faucet util function

### PR DESCRIPTION
**Pull request by AlbertSu123**

### CODE REVIEW

**Review by antiquicksort — APPROVED** [review]

File: src/common/constants.ts [hunk 1]
```diff
@@ -154,6 +154,8 @@ export const gasSpecificChain = {
     }
 }
 
+export const FUN_FAUCET_URL = "https://api.fun.xyz/demo-faucet"
```
Comment (Chazzzzzzz) [inline src/common/constants.ts:157]: use the API_URL constant here

File: src/utils/ChainUtils.ts [hunk 2]
```diff
@@ -118,3 +118,13 @@ export const getPermitHash = (token: Address, to: Address, amount: bigint, nonce
 export const getGasStation = async (gasStationUrl: string): Promise<any> => {
     return await sendRequest(gasStationUrl, "GET", "")
 }
+
+export const useFaucet = async (chain: Chain, wallet: FunWallet) => {
```
Comment (Chazzzzzzz) [inline src/utils/ChainUtils.ts:122]: create an api endpoint in apis folder

and then call that api here.

Comment (Chazzzzzzz) [reply src/utils/ChainUtils.ts:122]: useFaucet(chain: Chain|number|string, addr: String)

File: src/utils/ChainUtils.ts [hunk 3]
```diff
@@ -118,3 +118,13 @@ export const getPermitHash = (token: Address, to: Address, amount: bigint, nonce
...
+    const chainName = await chain.getChainName()
+    const walletAddress = await wallet.getAddress()
+    const ethRequest = await sendGetRequest(FUN_FAUCET_URL, `get-faucet?token=eth&testnet=${chainName}&addr=${walletAddress}`)
```
Comment (Chazzzzzzz) [inline src/utils/ChainUtils.ts:125]: throw error if chain not goerli

File: src/utils/ChainUtils.ts (hunk 2 above)
Comment (AlbertSu123) [reply src/utils/ChainUtils.ts:122]: resolved second, i don't think first is necessary

**Review by csirak1528 — APPROVED** [review]

**Review by Chazzzzzzz — APPROVED** [review]

**Pull request merged**